### PR TITLE
groups: fix rejoining a group after an error

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Fixes some edge cases in %group-view that would previously require a `|nuke %group-view` 

Additionally provides further interactivity if a group joining case fails. 
- Retry will simply attempt to rejoin the group
- Abort will remove the "Join Failed" tile and remove any persisted attempt, closing the window

Screen Shot 2022-01-31 at 4 24 15 pm
Screen Shot 2022-01-31 at 4 24 20 pm

### CODE REVIEW

**Review by arthyn — CHANGES_REQUESTED** [review]
Just a few small things otherwise looks good

File: pkg/interface/src/views/landscape/components/Join/Join.tsx [hunk 1]
```diff
@@ -150,22 +150,38 @@ function JoinError(props: {
...
   const group = preview?.metadata?.title ?? desc.group;
   const title = `Joining ${group} failed`;
   const explanation =
     request.progress === 'no-perms'
       ? 'You do not have the correct permissions'
-      : 'An unexpected error occurred';
+    : 'An unexpected error occurred';
+
+  const onRetry = () => {
+    useGroupState.getState().abortJoin(desc.group);
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/interface/src/views/landscape/components/Join/Join.tsx:165]: none that I can think of off the top of my head

File: pkg/interface/src/views/apps/launch/App.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export const LaunchApp = (props: LaunchAppProps): ReactElement | null => {
                 border={0}
                 p={0}
                 borderRadius={2}
-                onClick={() => history.push({ search: "?join-kind=group" })}
+                onClick={() => history.push({ search: "?join-kind=groups" })}
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/interface/src/views/apps/launch/App.tsx:97]: can't find that function in this checkout of `next/groups`?

[the remark this answers is not in the document]
Comment (arthyn) [reply pkg/interface/src/views/apps/launch/App.tsx:97]: Ah master probably needs to be merged into next/groups, sorry

[the remark this answers is not in the document]
Comment (arthyn) [reply pkg/interface/src/views/apps/launch/App.tsx:97]: Alright I just merged master into next/groups if you want to pull it in

[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply pkg/interface/src/views/apps/launch/App.tsx]: done

**Review by arthyn — APPROVED** [review]

**Comment by nerveharp** [comment]
Fixes urbit/landscape#1389

**Pull request merged**